src/ui/public/courier/data_source/_abstract.js

```diff
@@ -389,6 +391,8 @@ export default function SourceAbstractFactory(Private, Promise, PromiseEmitter)
             recurse(agg.aggs || agg.aggregations);
           });
         }(flatState.body.aggs || flatState.body.aggregations));
+
+        highlightBody(flatState.body);
```

Comment:
I like that moving highlighting to _flatten ensures that it works across discover and dashboard. But it also means that even Vis requests ask for highlighting. I wonder if we should try to avoid this? Since the highlighting logic is nicely modularized now maybe we could just reuse it in both discover and dashboard?

<img width="1509" alt="screen shot 2016-12-29 at 12 56 24 pm" src="https://cloud.githubusercontent.com/assets/6239176/21550323/d0ddd874-cdc6-11e6-803e-c90d689843f4.png">